<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>R Notebook</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Locus_Coruleus</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">R Notebook</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-05-31
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Locus_Coruleus/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220516code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220516)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220516code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220516)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongcca88ad">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> cca88ad
</a>
</p>
</div>
<div id="strongRepositoryversionstrongcca88ad"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version cca88ad.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/Seurat_proc_LC.nb.html
    Ignored:    analysis/test.nb.html

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Seurat_proc_LC.Rmd</code>) and
HTML (<code>docs/Seurat_proc_LC.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
cca88ad
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
<td>
wflow_publish("./analysis/Seurat_proc_LC.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
79c3bce
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
e407493
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
<td>
wflow_publish(".Rprofile", update = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
e407493
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
<td>
wflow_publish(".Rprofile", update = TRUE)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>This is an <a href="http://rmarkdown.rstudio.com">R Markdown</a>
Notebook. When you execute code within the notebook, the results appear
beneath the code.</p>
<p>Try executing this chunk by clicking the <em>Run</em> button within
the chunk or by placing your cursor inside it and pressing
<em>Ctrl+Shift+Enter</em>.</p>
<pre class="r"><code>rm(list = ls())
library(Seurat)</code></pre>
<pre><code>Attaching SeuratObject</code></pre>
<pre><code>Attaching sp</code></pre>
<pre class="r"><code>library(dplyr)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code>library(patchwork)
library(magrittr)
# Load the LC dataset
Loco.data &lt;- Read10X(data.dir = &quot;~/data/Locus_Coruleus2/count/&quot;)
# Initialize the Seurat object with the raw (non-normalized data).
Loco &lt;- CreateSeuratObject(counts = Loco.data, project = &quot;Locus_Coruleus&quot;, min.cells = 3, min.features = 200)
# The [[ operator can add columns to object metadata. This is a great place to stash QC stats
Loco[[&quot;percent.mt&quot;]] &lt;- PercentageFeatureSet(Loco, pattern = &quot;^mt-&quot;)
# Visualize QC metrics as a violin plot
VlnPlot(Loco, features = c(&quot;nFeature_RNA&quot;, &quot;nCount_RNA&quot;, &quot;percent.mt&quot;), ncol = 3)</code></pre>
<p><img src="figure/Seurat_proc_LC.Rmd/test1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-test1-1">
Past versions of test1-1.png
</button>
</p>
<div id="fig-test1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
79c3bce
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
<tr>
<td>
e407493
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># FeatureScatter is typically used to visualize feature-feature relationships, but can be used
# for anything calculated by the object, i.e. columns in object metadata, PC scores etc.</code></pre>
<pre class="r"><code>plot1 &lt;- FeatureScatter(Loco, feature1 = &quot;nCount_RNA&quot;, feature2 = &quot;percent.mt&quot;)
plot2 &lt;- FeatureScatter(Loco, feature1 = &quot;nCount_RNA&quot;, feature2 = &quot;nFeature_RNA&quot;)
plot1 + plot2</code></pre>
<p><img src="figure/Seurat_proc_LC.Rmd/test2-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-test2-1">
Past versions of test2-1.png
</button>
</p>
<div id="fig-test2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
79c3bce
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
<tr>
<td>
e407493
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Loco &lt;- subset(Loco, subset = nFeature_RNA &gt; 200 &amp; percent.mt &lt; 20)
# normalize and run dimensionality reduction on control dataset
Loco &lt;- SCTransform(Loco, return.only.var.genes = FALSE, verbose = FALSE) %&gt;%
    RunPCA(npcs = 100, verbose = FALSE) %&gt;%
    RunUMAP(dims = 1:20, verbose = FALSE) %&gt;%
    FindNeighbors(reduction = &quot;pca&quot;, dims = 1:100, n.trees = 200, annoy.metric = &quot;cosine&quot;, verbose = FALSE) %&gt;%
    FindClusters(resolution = 0.3, algorithm = 1, verbose = FALSE)</code></pre>
<pre><code>Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric
To use Python UMAP via reticulate, set umap.method to &#39;umap-learn&#39; and metric to &#39;correlation&#39;
This message will be shown once per session</code></pre>
<pre class="r"><code>DimHeatmap(Loco, dims = 1:20, cells = 500, balanced = TRUE)</code></pre>
<p><img src="figure/Seurat_proc_LC.Rmd/unnamed-chunk-1-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-1-1">
Past versions of unnamed-chunk-1-1.png
</button>
</p>
<div id="fig-unnamed-chunk-1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
79c3bce
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
<tr>
<td>
e407493
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ElbowPlot(Loco, ndims = 30)</code></pre>
<p><img src="figure/Seurat_proc_LC.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
79c3bce
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
<tr>
<td>
e407493
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(Loco, reduction = &quot;umap&quot;)</code></pre>
<p><img src="figure/Seurat_proc_LC.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
79c3bce
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
<tr>
<td>
e407493
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(Loco, features = c(&quot;Slc17a7&quot;), order = T)</code></pre>
<p><img src="figure/Seurat_proc_LC.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
79c3bce
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
<tr>
<td>
e407493
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Loco %&lt;&gt;%  FindClusters(resolution = 0.3, algorithm = 1, verbose = FALSE)
DimPlot(Loco, reduction = &quot;umap&quot;)</code></pre>
<p><img src="figure/Seurat_proc_LC.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
79c3bce
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
<tr>
<td>
e407493
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>head(Idents(Loco), 5)</code></pre>
<pre><code>AAACCCAAGACGATAT-1 AAACCCACAACCGCTG-1 AAACCCATCTTCTAAC-1 AAACGAAAGCGTGAAC-1 
                 0                  4                  0                  2 
AAACGAAGTGGTAACG-1 
                 0 
Levels: 0 1 2 3 4 5 6 7 8 9 10 11 12 13 14</code></pre>
<pre class="r"><code>Loco.markers &lt;- FindAllMarkers(Loco, only.pos = TRUE, test.use = &quot;wilcox&quot;, min.pct = 0.1, logfc.threshold = 0.25)</code></pre>
<pre><code>Calculating cluster 0</code></pre>
<pre><code>Calculating cluster 1</code></pre>
<pre><code>Calculating cluster 2</code></pre>
<pre><code>Calculating cluster 3</code></pre>
<pre><code>Calculating cluster 4</code></pre>
<pre><code>Calculating cluster 5</code></pre>
<pre><code>Calculating cluster 6</code></pre>
<pre><code>Calculating cluster 7</code></pre>
<pre><code>Calculating cluster 8</code></pre>
<pre><code>Calculating cluster 9</code></pre>
<pre><code>Calculating cluster 10</code></pre>
<pre><code>Calculating cluster 11</code></pre>
<pre><code>Calculating cluster 12</code></pre>
<pre><code>Calculating cluster 13</code></pre>
<pre><code>Calculating cluster 14</code></pre>
<pre class="r"><code>Loco.markers %&gt;%
    group_by(cluster) %&gt;%
    slice_max(n = 10, order_by = avg_log2FC)</code></pre>
<pre><code># A tibble: 150 × 7
# Groups:   cluster [15]
       p_val avg_log2FC pct.1 pct.2 p_val_adj cluster gene   
       &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt; &lt;fct&gt;   &lt;chr&gt;  
 1 0               3.77 1     0.416 0         0       Plp1   
 2 0               2.67 0.993 0.374 0         0       Pde4b  
 3 0               2.55 0.972 0.109 0         0       Mbp    
 4 0               2.39 0.965 0.246 0         0       Tmeff2 
 5 0               2.23 1     0.824 0         0       Pcdh9  
 6 0               2.07 0.985 0.472 0         0       Nkain2 
 7 0               2.02 0.929 0.188 0         0       Rnf220 
 8 0               1.99 0.962 0.29  0         0       Slc24a2
 9 1.10e-306       1.97 0.929 0.251 1.81e-302 0       Plcl1  
10 0               1.94 0.907 0.151 0         0       Dock10 
# … with 140 more rows</code></pre>
<pre class="r"><code>Locon &lt;- subset(Loco, idents = c(1,4,7,8,10,11,12,13,14))
    RunPCA(Locon, npcs = 30, verbose = FALSE) %&gt;%
    RunUMAP(dims = 1:30, verbose = FALSE) %&gt;%
    FindNeighbors(reduction = &quot;pca&quot;, dims = 1:30, n.trees = 200, annoy.metric = &quot;cosine&quot;, verbose = FALSE) %&gt;%
    FindClusters(resolution = 0.3, algorithm = 1, verbose = FALSE)</code></pre>
<pre><code>An object of class Seurat 
34320 features across 1151 samples within 2 assays 
Active assay: SCT (16437 features, 3000 variable features)
 1 other assay present: RNA
 2 dimensional reductions calculated: pca, umap</code></pre>
<pre class="r"><code>DimHeatmap(Locon, dims = 1:20, cells = 500, balanced = TRUE)</code></pre>
<p><img src="figure/Seurat_proc_LC.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
79c3bce
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
<tr>
<td>
e407493
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(Locon, reduction = &quot;umap&quot;)</code></pre>
<p><img src="figure/Seurat_proc_LC.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
79c3bce
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
<tr>
<td>
e407493
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Locon2 &lt;- SCTransform(Locon, return.only.var.genes = FALSE, verbose = FALSE) %&gt;%
    RunPCA(npcs = 30, verbose = FALSE) %&gt;%
    RunUMAP( dims = 1:30, verbose = FALSE) %&gt;%
    FindNeighbors(reduction = &quot;pca&quot;, dims = 1:30, n.trees = 200, annoy.metric = &quot;cosine&quot;, verbose = FALSE) %&gt;%
    FindClusters(resolution = 0.3, algorithm = 1, verbose = FALSE)
DimHeatmap(Locon2, dims = 1:20, cells = 500, balanced = TRUE)</code></pre>
<p><img src="figure/Seurat_proc_LC.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
79c3bce
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
<tr>
<td>
e407493
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(Locon2, reduction = &quot;umap&quot;)</code></pre>
<p><img src="figure/Seurat_proc_LC.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
79c3bce
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
<tr>
<td>
e407493
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(Locon2, features = c(&quot;Gal&quot;, &quot;Grp&quot;, &quot;Pdyn&quot;, &quot;Cartpt&quot;, &quot;Bdnf&quot;, &quot;Hcrtr1&quot;, &quot;Eya2&quot;, &quot;Th&quot;, &quot;Dbh&quot;), order = T)</code></pre>
<p><img src="figure/Seurat_proc_LC.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
79c3bce
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
<tr>
<td>
e407493
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>LoconTH &lt;- subset(Loco, idents = 8)
LoconTH &lt;- FindNeighbors(LoconTH, reduction = &quot;pca&quot;, dims = 1:30, n.trees = 200, annoy.metric = &quot;cosine&quot;, verbose = FALSE) %&gt;%
    FindClusters(resolution = 0.3, algorithm = 1, verbose = FALSE)%&gt;%
    RunUMAP( dims = 1:30, verbose = FALSE) 
DimHeatmap(LoconTH, dims = 1:20, cells = 500, balanced = TRUE)</code></pre>
<pre><code>Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.</code></pre>
<p><img src="figure/Seurat_proc_LC.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
79c3bce
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
<tr>
<td>
e407493
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ElbowPlot(LoconTH, ndims = 30)</code></pre>
<p><img src="figure/Seurat_proc_LC.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
79c3bce
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
<tr>
<td>
e407493
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(LoconTH, reduction = &quot;umap&quot;)</code></pre>
<p><img src="figure/Seurat_proc_LC.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
79c3bce
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
<tr>
<td>
e407493
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(LoconTH, features = c(&quot;Gal&quot;, &quot;Grp&quot;, &quot;Pdyn&quot;, &quot;Cartpt&quot;, &quot;Bdnf&quot;, &quot;Th&quot;), order = T)</code></pre>
<p><img src="figure/Seurat_proc_LC.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
79c3bce
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
<tr>
<td>
e407493
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Locon3 &lt;- subset(Locon2, idents = 5)
    RunPCA(Locon2, npcs = 30, verbose = FALSE) %&gt;%
    RunUMAP(dims = 1:30, verbose = FALSE) %&gt;%
    FindNeighbors(reduction = &quot;pca&quot;, dims = 1:30, n.trees = 200, annoy.metric = &quot;cosine&quot;, verbose = FALSE) %&gt;%
    FindClusters(resolution = 0.3, algorithm = 1, verbose = FALSE)</code></pre>
<pre><code>An object of class Seurat 
32614 features across 1151 samples within 2 assays 
Active assay: SCT (14731 features, 3000 variable features)
 1 other assay present: RNA
 2 dimensional reductions calculated: pca, umap</code></pre>
<pre class="r"><code>DimHeatmap(Locon3, dims = 1:20, cells = 500, balanced = TRUE)</code></pre>
<pre><code>Warning: Requested number is larger than the number of available items (82).
Setting to 82.

Warning: Requested number is larger than the number of available items (82).
Setting to 82.

Warning: Requested number is larger than the number of available items (82).
Setting to 82.

Warning: Requested number is larger than the number of available items (82).
Setting to 82.

Warning: Requested number is larger than the number of available items (82).
Setting to 82.

Warning: Requested number is larger than the number of available items (82).
Setting to 82.

Warning: Requested number is larger than the number of available items (82).
Setting to 82.

Warning: Requested number is larger than the number of available items (82).
Setting to 82.

Warning: Requested number is larger than the number of available items (82).
Setting to 82.

Warning: Requested number is larger than the number of available items (82).
Setting to 82.

Warning: Requested number is larger than the number of available items (82).
Setting to 82.

Warning: Requested number is larger than the number of available items (82).
Setting to 82.

Warning: Requested number is larger than the number of available items (82).
Setting to 82.

Warning: Requested number is larger than the number of available items (82).
Setting to 82.

Warning: Requested number is larger than the number of available items (82).
Setting to 82.

Warning: Requested number is larger than the number of available items (82).
Setting to 82.

Warning: Requested number is larger than the number of available items (82).
Setting to 82.

Warning: Requested number is larger than the number of available items (82).
Setting to 82.

Warning: Requested number is larger than the number of available items (82).
Setting to 82.

Warning: Requested number is larger than the number of available items (82).
Setting to 82.</code></pre>
<p><img src="figure/Seurat_proc_LC.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
79c3bce
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
<tr>
<td>
e407493
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(Locon3, features = c(&quot;Gal&quot;, &quot;Grp&quot;, &quot;Pdyn&quot;, &quot;Cartpt&quot;, &quot;Bdnf&quot;, &quot;Th&quot;, &quot;Slc18a2&quot;, &quot;Dbh&quot;), order = T)</code></pre>
<p><img src="figure/Seurat_proc_LC.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
79c3bce
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
<tr>
<td>
e407493
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Locon3th &lt;- subset(Locon3, Slc18a2 &gt; 1)</code></pre>
<pre class="r"><code>FeaturePlot(Locon3th, features = c(&quot;Gal&quot;, &quot;Grp&quot;, &quot;Pdyn&quot;, &quot;Cartpt&quot;, &quot;Bdnf&quot;, &quot;Th&quot;, &quot;Slc18a2&quot;), order = T)</code></pre>
<p><img src="figure/Seurat_proc_LC.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
79c3bce
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
<tr>
<td>
e407493
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>Locon4 &lt;- FindNeighbors(Locon, reduction = &quot;pca&quot;, dims = 1:30, n.trees = 200, annoy.metric = &quot;cosine&quot;, verbose = FALSE) %&gt;%
    FindClusters(resolution = 0.3, algorithm = 1, verbose = FALSE) %&gt;%
    RunUMAP( dims = 1:30, verbose = FALSE)
DimHeatmap(Locon4, dims = 1:20, cells = 500, balanced = TRUE)</code></pre>
<p><img src="figure/Seurat_proc_LC.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
79c3bce
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
<tr>
<td>
e407493
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(Locon4, reduction = &quot;umap&quot;)</code></pre>
<p><img src="figure/Seurat_proc_LC.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-1">
Past versions of unnamed-chunk-22-1.png
</button>
</p>
<div id="fig-unnamed-chunk-22-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
79c3bce
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
<tr>
<td>
e407493
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>FeaturePlot(Locon4, features = c(&quot;Gal&quot;, &quot;Grp&quot;, &quot;Sst&quot;, &quot;Slc17a6&quot;, &quot;Pdyn&quot;, &quot;Cartpt&quot;, &quot;Bdnf&quot;, &quot;Th&quot;, &quot;Slc18a2&quot;, &quot;Dbh&quot;, &quot;Gad1&quot;), order = T)</code></pre>
<p><img src="figure/Seurat_proc_LC.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
79c3bce
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
<tr>
<td>
e407493
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>SeuratDisk::SaveH5Seurat(Locon4, 
                         filename = paste0(&quot;./data/&quot;,
                                           Project(object = Locon4), 
                                           &quot;.h5Seurat&quot;), 
                         overwrite = TRUE)</code></pre>
<pre><code>Registered S3 method overwritten by &#39;SeuratDisk&#39;:
  method            from  
  as.sparse.H5Group Seurat</code></pre>
<pre><code>Creating h5Seurat file for version 3.1.5.9900</code></pre>
<pre><code>Adding counts for RNA</code></pre>
<pre><code>Adding data for RNA</code></pre>
<pre><code>No variable features found for RNA</code></pre>
<pre><code>No feature-level metadata found for RNA</code></pre>
<pre><code>Adding counts for SCT</code></pre>
<pre><code>Adding data for SCT</code></pre>
<pre><code>Adding scale.data for SCT</code></pre>
<pre><code>Adding variable features for SCT</code></pre>
<pre><code>No feature-level metadata found for SCT</code></pre>
<pre><code>Writing out SCTModel.list for SCT</code></pre>
<pre><code>Adding cell embeddings for pca</code></pre>
<pre><code>Adding loadings for pca</code></pre>
<pre><code>No projected loadings for pca</code></pre>
<pre><code>Adding standard deviations for pca</code></pre>
<pre><code>No JackStraw data for pca</code></pre>
<pre><code>Adding cell embeddings for umap</code></pre>
<pre><code>No loadings for umap</code></pre>
<pre><code>No projected loadings for umap</code></pre>
<pre><code>No standard deviations for umap</code></pre>
<pre><code>No JackStraw data for umap</code></pre>
<pre class="r"><code>LoconTH4 &lt;- subset(Locon4, idents = 3)
LoconTH4 &lt;- FindNeighbors(LoconTH4, reduction = &quot;pca&quot;, dims = 1:30, n.trees = 200, annoy.metric = &quot;cosine&quot;, verbose = FALSE) %&gt;%
    FindClusters(resolution = 0.3, algorithm = 1, verbose = FALSE)%&gt;%
    RunUMAP( dims = 1:30, verbose = FALSE) 
DimHeatmap(LoconTH4, dims = 1:20, cells = 500, balanced = TRUE)</code></pre>
<pre><code>Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.

Warning: Requested number is larger than the number of available items (80).
Setting to 80.</code></pre>
<p><img src="figure/Seurat_proc_LC.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
79c3bce
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
<tr>
<td>
e407493
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(LoconTH4, reduction = &quot;umap&quot;)</code></pre>
<p><img src="figure/Seurat_proc_LC.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-1">
Past versions of unnamed-chunk-25-1.png
</button>
</p>
<div id="fig-unnamed-chunk-25-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
79c3bce
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
<tr>
<td>
e407493
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>DimPlot(Locon4, reduction = “umap”)</p>
<pre class="r"><code>FeaturePlot(LoconTH4, slot = &quot;scale.data&quot;, features = c(&quot;Gal&quot;, &quot;Grp&quot;, &quot;Pdyn&quot;, &quot;Cartpt&quot;, &quot;Bdnf&quot;, &quot;Calca&quot;, &quot;Hcrtr1&quot;, &quot;Sstr2&quot;, &quot;Calcr&quot;, &quot;Slc6a2&quot;, &quot;Slc18a2&quot;, &quot;Dbh&quot;), order = T)</code></pre>
<p><img src="figure/Seurat_proc_LC.Rmd/unnamed-chunk-26-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-26-1">
Past versions of unnamed-chunk-26-1.png
</button>
</p>
<div id="fig-unnamed-chunk-26-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
79c3bce
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
<tr>
<td>
e407493
</td>
<td>
cheglok81
</td>
<td>
2022-05-31
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>LoconTH4.markers &lt;- FindAllMarkers(LoconTH4, only.pos = TRUE, test.use = &quot;MAST&quot;, min.pct = 0.1, logfc.threshold = 0.25)</code></pre>
<pre><code>Calculating cluster 0</code></pre>
<pre><code>
Done!</code></pre>
<pre><code>Combining coefficients and standard errors</code></pre>
<pre><code>Calculating log-fold changes</code></pre>
<pre><code>Calculating likelihood ratio tests</code></pre>
<pre><code>Refitting on reduced model...</code></pre>
<pre><code>
Done!</code></pre>
<pre><code>Calculating cluster 1</code></pre>
<pre><code>
Done!</code></pre>
<pre><code>Combining coefficients and standard errors</code></pre>
<pre><code>Calculating log-fold changes</code></pre>
<pre><code>Calculating likelihood ratio tests</code></pre>
<pre><code>Refitting on reduced model...</code></pre>
<pre><code>
Done!</code></pre>
<pre class="r"><code>LoconTH4.markers %&gt;%
    group_by(cluster) %&gt;%
    slice_max(n = 10, order_by = avg_log2FC)</code></pre>
<pre><code># A tibble: 20 × 7
# Groups:   cluster [2]
      p_val avg_log2FC pct.1 pct.2  p_val_adj cluster gene   
      &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;      &lt;dbl&gt; &lt;fct&gt;   &lt;chr&gt;  
 1 8.96e-10      1.42  1     0.88  0.0000147  0       Slc18a2
 2 9.22e-10      1.12  0.927 0.4   0.0000152  0       Morf4l1
 3 7.88e-10      1.10  0.964 0.4   0.0000129  0       Prnp   
 4 3.12e- 8      1.09  0.945 0.56  0.000512   0       Atp1b1 
 5 2.13e- 6      0.968 0.927 0.52  0.0351     0       Nap1l5 
 6 3.01e- 6      0.955 0.818 0.36  0.0495     0       Dlk1   
 7 1.22e- 6      0.927 0.891 0.48  0.0200     0       Ywhab  
 8 3.25e- 7      0.926 0.727 0.12  0.00534    0       Arl6ip1
 9 6.36e- 9      0.923 0.927 0.28  0.000105   0       Actb   
10 6.79e- 7      0.897 0.818 0.24  0.0112     0       Tmem30a
11 2.93e- 4      1.14  0.72  0.364 1          1       Robo1  
12 1.52e- 8      0.975 1     0.891 0.000249   1       Frmd4a 
13 2.98e- 7      0.917 1     0.909 0.00490    1       Lingo2 
14 7.91e-11      0.896 1     1     0.00000130 1       Sdk1   
15 2.42e- 9      0.877 1     0.855 0.0000399  1       Csmd1  
16 1.26e-10      0.855 1     1     0.00000208 1       Nrg3   
17 2.31e- 8      0.832 1     1     0.000379   1       Lsamp  
18 2.01e- 5      0.822 0.96  0.927 0.330      1       Galntl6
19 2.25e- 5      0.793 0.76  0.418 0.369      1       Magi2  
20 3.11e- 6      0.782 0.92  0.945 0.0511     1       Ank3   </code></pre>
<pre class="r"><code>LoconTH4.markers &lt;- FindAllMarkers(LoconTH4, only.pos = TRUE, test.use = &quot;LR&quot;)</code></pre>
<pre><code>Calculating cluster 0</code></pre>
<pre><code>Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred

Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred

Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred

Warning: glm.fit: fitted probabilities numerically 0 or 1 occurred</code></pre>
<pre><code>Calculating cluster 1</code></pre>
<pre class="r"><code>LoconTH4.markers %&gt;%
    group_by(cluster) %&gt;%
    slice_max(n = 10, order_by = avg_log2FC)</code></pre>
<pre><code># A tibble: 20 × 7
# Groups:   cluster [2]
      p_val avg_log2FC pct.1 pct.2    p_val_adj cluster gene   
      &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;        &lt;dbl&gt; &lt;fct&gt;   &lt;chr&gt;  
 1 3.97e-10      1.42  1     0.88  0.00000653   0       Slc18a2
 2 1.52e-10      1.12  0.927 0.4   0.00000250   0       Morf4l1
 3 1.99e-11      1.10  0.964 0.4   0.000000328  0       Prnp   
 4 1.03e- 8      1.09  0.945 0.56  0.000169     0       Atp1b1 
 5 2.17e- 7      0.968 0.927 0.52  0.00357      0       Nap1l5 
 6 5.71e- 7      0.955 0.818 0.36  0.00938      0       Dlk1   
 7 2.07e- 7      0.927 0.891 0.48  0.00341      0       Ywhab  
 8 3.60e- 8      0.926 0.727 0.12  0.000592     0       Arl6ip1
 9 8.98e-10      0.923 0.927 0.28  0.0000148    0       Actb   
10 5.49e- 8      0.897 0.818 0.24  0.000902     0       Tmem30a
11 6.69e- 5      1.14  0.72  0.364 1            1       Robo1  
12 7.18e- 9      0.975 1     0.891 0.000118     1       Frmd4a 
13 4.55e- 8      0.917 1     0.909 0.000747     1       Lingo2 
14 4.70e-12      0.896 1     1     0.0000000773 1       Sdk1   
15 6.02e-10      0.877 1     0.855 0.00000989   1       Csmd1  
16 3.12e-12      0.855 1     1     0.0000000513 1       Nrg3   
17 1.19e- 9      0.832 1     1     0.0000195    1       Lsamp  
18 3.36e- 5      0.822 0.96  0.927 0.553        1       Galntl6
19 2.29e- 5      0.793 0.76  0.418 0.376        1       Magi2  
20 1.59e- 4      0.782 0.92  0.945 1            1       Ank3   </code></pre>
<pre class="r"><code>LoconTH4.markers %&gt;% filter(gene == &quot;Grp&quot;)</code></pre>
<pre><code>[1] p_val      avg_log2FC pct.1      pct.2      p_val_adj  cluster    gene      
&lt;0 rows&gt; (or 0-length row.names)</code></pre>
<pre class="r"><code>LoconTH4.markers &lt;- FindAllMarkers(LoconTH4, only.pos = TRUE, test.use = &quot;roc&quot;, min.pct = 0.1)</code></pre>
<pre><code>Calculating cluster 0</code></pre>
<pre><code>Calculating cluster 1</code></pre>
<pre class="r"><code>LoconTH4.markers %&gt;%
    group_by(cluster) %&gt;%
    slice_max(n = 10, order_by = avg_log2FC)</code></pre>
<pre><code># A tibble: 20 × 8
# Groups:   cluster [2]
   myAUC avg_diff power avg_log2FC pct.1 pct.2 cluster gene   
   &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt;      &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;fct&gt;   &lt;chr&gt;  
 1 0.902    0.985 0.804      1.42  1     0.88  0       Slc18a2
 2 0.898    0.774 0.796      1.12  0.927 0.4   0       Morf4l1
 3 0.902    0.764 0.804      1.10  0.964 0.4   0       Prnp   
 4 0.871    0.754 0.742      1.09  0.945 0.56  0       Atp1b1 
 5 0.829    0.671 0.658      0.968 0.927 0.52  0       Nap1l5 
 6 0.817    0.662 0.634      0.955 0.818 0.36  0       Dlk1   
 7 0.837    0.643 0.674      0.927 0.891 0.48  0       Ywhab  
 8 0.824    0.642 0.648      0.926 0.727 0.12  0       Arl6ip1
 9 0.864    0.640 0.728      0.923 0.927 0.28  0       Actb   
10 0.831    0.622 0.662      0.897 0.818 0.24  0       Tmem30a
11 0.732    0.788 0.464      1.14  0.72  0.364 1       Robo1  
12 0.853    0.676 0.706      0.975 1     0.891 1       Frmd4a 
13 0.844    0.635 0.688      0.917 1     0.909 1       Lingo2 
14 0.926    0.621 0.852      0.896 1     1     1       Sdk1   
15 0.871    0.608 0.742      0.877 1     0.855 1       Csmd1  
16 0.922    0.593 0.844      0.855 1     1     1       Nrg3   
17 0.879    0.577 0.758      0.832 1     1     1       Lsamp  
18 0.769    0.570 0.538      0.822 0.96  0.927 1       Galntl6
19 0.753    0.550 0.506      0.793 0.76  0.418 1       Magi2  
20 0.768    0.542 0.536      0.782 0.92  0.945 1       Ank3   </code></pre>
<p>Add a new chunk by clicking the <em>Insert Chunk</em> button on the
toolbar or by pressing <em>Ctrl+Alt+I</em>.When you save the notebook,
an HTML file containing the code and output will be saved alongside it
(click the <em>Preview</em> button or press <em>Ctrl+Shift+K</em> to
preview the HTML file).</p>
<p>The preview shows you a rendered HTML copy of the contents of the
editor. Consequently, unlike <em>Knit</em>, <em>Preview</em> does not
run any R code chunks. Instead, the output of the chunk when it was last
run in the editor is displayed.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] magrittr_2.0.3     patchwork_1.1.1    dplyr_1.0.9        sp_1.4-7          
[5] SeuratObject_4.1.0 Seurat_4.1.1       workflowr_1.7.0   

loaded via a namespace (and not attached):
  [1] plyr_1.8.7                  igraph_1.3.1               
  [3] lazyeval_0.2.2              splines_4.2.0              
  [5] listenv_0.8.0               scattermore_0.8            
  [7] GenomeInfoDb_1.32.2         ggplot2_3.3.6              
  [9] digest_0.6.29               htmltools_0.5.2            
 [11] fansi_1.0.3                 tensor_1.5                 
 [13] cluster_2.1.3               ROCR_1.0-11                
 [15] limma_3.52.1                globals_0.15.0             
 [17] matrixStats_0.62.0          spatstat.sparse_2.1-1      
 [19] prettyunits_1.1.1           colorspace_2.0-3           
 [21] ggrepel_0.9.1.9999          xfun_0.31                  
 [23] rgdal_1.5-32                RCurl_1.98-1.6             
 [25] callr_3.7.0                 crayon_1.5.1               
 [27] jsonlite_1.8.0              progressr_0.10.0           
 [29] spatstat.data_2.2-0         survival_3.3-1             
 [31] zoo_1.8-10                  glue_1.6.2                 
 [33] polyclip_1.10-0             gtable_0.3.0               
 [35] zlibbioc_1.42.0             XVector_0.36.0             
 [37] leiden_0.4.2                DelayedArray_0.22.0        
 [39] SingleCellExperiment_1.18.0 future.apply_1.9.0         
 [41] BiocGenerics_0.42.0         abind_1.4-5                
 [43] scales_1.2.0                DBI_1.1.2                  
 [45] spatstat.random_2.2-0       miniUI_0.1.1.1             
 [47] Rcpp_1.0.8.3                progress_1.2.2             
 [49] viridisLite_0.4.0           xtable_1.8-4               
 [51] reticulate_1.25             spatstat.core_2.4-4        
 [53] bit_4.0.4                   stats4_4.2.0               
 [55] htmlwidgets_1.5.4           httr_1.4.3                 
 [57] RColorBrewer_1.1-3          ellipsis_0.3.2             
 [59] ica_1.0-2                   pkgconfig_2.0.3            
 [61] farver_2.1.0                sass_0.4.1                 
 [63] uwot_0.1.11                 deldir_1.0-6               
 [65] utf8_1.2.2                  tidyselect_1.1.2           
 [67] labeling_0.4.2              rlang_1.0.2                
 [69] reshape2_1.4.4              later_1.3.0                
 [71] munsell_0.5.0               tools_4.2.0                
 [73] cli_3.3.0                   generics_0.1.2             
 [75] ggridges_0.5.3              evaluate_0.15              
 [77] stringr_1.4.0               fastmap_1.1.0              
 [79] yaml_2.3.5                  goftest_1.2-3              
 [81] bit64_4.0.5                 processx_3.5.3             
 [83] knitr_1.39                  fs_1.5.2                   
 [85] fitdistrplus_1.1-8          purrr_0.3.4                
 [87] RANN_2.6.1                  pbapply_1.5-0              
 [89] future_1.26.1               nlme_3.1-157               
 [91] whisker_0.4                 mime_0.12                  
 [93] ggrastr_1.0.1               hdf5r_1.3.5                
 [95] compiler_4.2.0              rstudioapi_0.13            
 [97] beeswarm_0.4.0              plotly_4.10.0              
 [99] png_0.1-7                   spatstat.utils_2.3-1       
[101] tibble_3.1.7                bslib_0.3.1                
[103] stringi_1.7.6               highr_0.9                  
[105] ps_1.7.0                    RSpectra_0.16-1            
[107] rgeos_0.5-9                 lattice_0.20-45            
[109] Matrix_1.4-1                SeuratDisk_0.0.0.9020      
[111] vctrs_0.4.1                 pillar_1.7.0               
[113] lifecycle_1.0.1             spatstat.geom_2.4-0        
[115] lmtest_0.9-40               jquerylib_0.1.4            
[117] RcppAnnoy_0.0.19            bitops_1.0-7               
[119] data.table_1.14.2           cowplot_1.1.1              
[121] irlba_2.3.5                 GenomicRanges_1.48.0       
[123] httpuv_1.6.5                R6_2.5.1                   
[125] promises_1.2.0.1            KernSmooth_2.23-20         
[127] gridExtra_2.3               IRanges_2.30.0             
[129] vipor_0.4.5                 parallelly_1.31.1          
[131] codetools_0.2-18            MASS_7.3-57                
[133] assertthat_0.2.1            SummarizedExperiment_1.26.1
[135] MAST_1.22.0                 rprojroot_2.0.3            
[137] withr_2.5.0                 sctransform_0.3.3          
[139] GenomeInfoDbData_1.2.8      S4Vectors_0.34.0           
[141] hms_1.1.1                   mgcv_1.8-40                
[143] parallel_4.2.0              grid_4.2.0                 
[145] rpart_4.1.16                tidyr_1.2.0                
[147] rmarkdown_2.14              MatrixGenerics_1.8.0       
[149] Rtsne_0.16                  git2r_0.30.1               
[151] getPass_0.2-2               Biobase_2.56.0             
[153] shiny_1.7.1                 ggbeeswarm_0.6.0           </code></pre>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
